sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/MLIRExporter.cs

```diff
@@ -0,0 +1,853 @@
+            writer.WriteLine($"cbde.store %{OpId(assignExpr)}, %{OpId(decl)} : memref<{MLIRType(assignExpr)}> {GetLocation(op)}");
+        }
+
+        private void ExportPrePostIncrementDecrement(ExpressionSyntax op, SyntaxToken opToken, ExpressionSyntax operand, bool isPostOperation)
+        {
+            // For now we only handle IdentifierNameSyntax (not ElementAccessExpressionSyntax or other)
+            var id = operand as IdentifierNameSyntax;
+            if (null == id)
+            {
+                writer.WriteLine($"%{OpId(op)} = cbde.unknown : {MLIRType(operand)} {GetLocation(op)} // Inc/Decrement of unknown identifier");
+                return;
+            }
+
+            var declSymbol = semanticModel.GetSymbolInfo(id);
+            if (!IsSymbolSupportedForAssignment(declSymbol) || !AreTypesSupported(operand))
+            {
+                writer.WriteLine($"%{OpId(op)} = cbde.unknown : {MLIRType(operand)} {GetLocation(op)} // Inc/Decrement of field or property {id.Identifier.ValueText}");
+                return;
+            }
+
+            var decl = declSymbol.Symbol.DeclaringSyntaxReferences[0].GetSyntax();
+            if (isPostOperation)
+            {
+                opMap[op] = opMap[operand];
+            }
+
+            var newCstId = UniqueOpId();
+            writer.WriteLine($"%{newCstId} = constant 1 : {MLIRType(operand)} {GetLocation(op)}");
+
+            var opId = isPostOperation ? UniqueOpId() : OpId(op);
+            var opName = opToken.IsKind(SyntaxKind.PlusPlusToken) ? "addi" : "subi";
+            writer.WriteLine($"%{opId} = {opName} %{OpId(operand)}, %{newCstId} : {MLIRType(operand)} {GetLocation(op)}");
+
+            writer.WriteLine($"cbde.store %{opId}, %{OpId(decl)} : memref<{MLIRType(operand)}> {GetLocation(op)}");
+        }
+
+        private bool AreTypesSupported(params ExpressionSyntax [] exprs)
+        {
+            return exprs.All(expr => IsTypeKnown(semanticModel.GetTypeInfo(expr).Type));
+        }
+
+        private void ExportComparison(string compName, SyntaxNode op)
+        {
+            var binExpr = op as BinaryExpressionSyntax;
+            if (!AreTypesSupported(binExpr.Left, binExpr.Right))
+            {
+                writer.WriteLine($"%{OpId(op)} = cbde.unknown : i1  {GetLocation(binExpr)} // comparison of unknown type: {op.Dump()}");
+                return;
+            }
+            // The type is the type of the operands, not of the result, which is always i1
+            writer.WriteLine($"%{OpId(op)} = cmpi \"{compName}\", %{OpId(getAssignmentValue(binExpr.Left))}, %{OpId(getAssignmentValue(binExpr.Right))} : {MLIRType(binExpr.Left)} {GetLocation(binExpr)}");
+        }
+
+        private string GetLocation(SyntaxNode node)
+        {
+            if (!exportsLocations)
+            {
+                return string.Empty;
+            }
+            // TODO: We should decide which of GetLineSpan or GetMappedLineSpan is better to use
```

Comment:
please open a github issue for this

Reply:
or, as discussed with @loic-joly-sonarsource offline, add the EPIC/MMF jira id in the comment, to be able to see if it's still in progress or already done